modules/tensor_mechanics/doc/content/documentation/modules/tensor_mechanics/C0TimoshenkoBeam.md

```diff
@@ -0,0 +1,161 @@
+
+If displacement or rotational eigenstrains are provided as input, those eigenstrain increments would be subtracted from the total displacement and rotational strain increments to get the mechanical displacement and rotational strain increments.
+
+## Stress calculation
+
+The axial and shear strain increments, and the rotational strain increments (after removal of eigenstrains) are passed to [ComputeBeamResultants](/ComputeBeamResultants.md) to calculate the force ($\Delta F$) and moment ($\Delta M$) increments using the linear elastic constitutive relations defined in [ComputeElasticityBeam](/ComputeElasticityBeam.md) as follows:
+\begin{equation}
+\Delta F_1(x) = E \; \epsilon_{1}(x)
+\end{equation}
+\begin{equation}
+\Delta F_2(x) = kG \; \epsilon_{2}(x)
+\end{equation}
+\begin{equation}
+\Delta F_3(x) = kG \; \epsilon_{3}(x)
+\end{equation}
+\begin{equation}
+\Delta M_1(x) = kG \; \kappa_1(x)
+\end{equation}
+\begin{equation}
+\Delta M_2(x) = E \; \kappa_2(x)
+\end{equation}
+\begin{equation}
+\Delta M_3(x) = G \; \kappa_3(x)
+\end{equation}
+where, $E$ and $G$ are the Young's and shear modulus, respectively. $k$ is the shear correction factor that depends on the shape of the cross-section.
+
+Since the strain increments are in the beam local configuration at $t+\Delta t$, the calculated force and moment increments are also in the same configuration. To get the total force and moment at $t + \Delta t$, the force and moment increments are first transformed to the global coordinate system using ${}^{t+\Delta t}R$ and then added to the forces and moments from $t$ that are already in the global coordinate system.
+
+The strain energy is calculated as:
+\begin{equation}
+\delta V = \int_\Omega (\sigma_{11}(x,y,z) \delta \epsilon_{11}(x,y,z) + \sigma_{12}(x,y,z) \delta \epsilon_{12}(x,y,z) + \sigma_{13}(x,y,z) \delta \epsilon_{13}(x,y,z)) \delta \Omega
+\end{equation}
+\begin{equation}
+= \int_{0}^{{}^0L} F_1(x) \delta \epsilon_{1}(x) + F_2(x) \delta \epsilon_{2}(x) + F_3(x) \delta \epsilon_{3}(x) + M_1(x) \delta \kappa_1(x) + M_2(x) \delta \kappa_2(x) + M_3(x) \delta \kappa_3(x)
+\end{equation}
+
+As can be seen from the equation for $\epsilon_{1}(x)$, it depends only on $\frac{\partial u_1}{\partial x}$ if $A_y$ and $A_z$ are zero. But $\epsilon_{2}(x)$ depends on both $\frac{\partial u_1}{\partial x}$ and $\theta_3$. Therefore, $F_2(x) \delta \epsilon_{2}(x)$ would contribute to the residual of both the translational DOF in y direction and rotational DOF in the z direction. Similarly, $F_3(x) \delta \epsilon_{3}(x)$ would contribute to the residual of both the translational DOF in z direction and rotational DOF in the y direction. This accounts for the coupling between the shear and bending behavior of the beam. The stress-divergence calculation is performed in [StressDivergenceBeam](/StressDivergenceBeam.md).
```

Comment:
@sveerara I think you're missing the references section here. The bibliography is not showing up.

Reply:
@cbolisetti, thanks for letting me know. I have added the bibliography line to the .md file. Surprisingly none of the tensor mechanics files have in documentation/modules/tensor_mechanics have a references section. Looks like the bibliography line somehow got dropped when it was moved to the new documentation system. 

Also, thanks for pointing out that `y_orientation` is not normalized within ComputeIncrementalBeamStrain.C. I have corrected that now.